### PR TITLE
Add Plugins: custom distribute silver

### PR DESCRIPTION
**Pull request by thickforest**
Hi,   I have written a new plugin for custom distribute silver.

The 'silver-spacetime' plugins has done a lot of automated works, while sometime asteroid is too far away from spacetime rip, use this plugin, players can do some customized configuration, for example, asteroid first send silver to a planet, and then send the silver to space rip.

Hopefully this plugin is useful.

### CODE REVIEW

**Comment by thickforest** [comment]
Hi, @jacobrosenthal

**Comment by jacobrosenthal** [comment]
Youll want to get feedback from the community before you ping me. Make sure to post in the plugins discord and have your friends check it out and report back

File: content/productivity/custom-distribute-silver/index.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+title: Custom Distribute Silver
+date: 2021-10-03T21:03:00
+subtitle: Custom distribute your silver!
+version: 0.0.1
```
Comment (jacobrosenthal) [inline content/productivity/custom-distribute-silver/index.md:5]: Its a little confusing but we use version to say which dark forest api version the plugin is targetting. So this should be 
```suggestion
version: 0.6.4
```

File: content/productivity/custom-distribute-silver/plugin.js [hunk 2]
```diff
@@ -0,0 +1,444 @@
+// Custom distribute Silver
```
Comment (jacobrosenthal) [inline content/productivity/custom-distribute-silver/plugin.js:1]: Can you say why its different from the other silver plugin here. Also state the constants and defaults that people can change. " By default sends a min of 2000 silver to a concurrent 5 ... all configurable by changing the constants at the top of the file"

Comment (jacobrosenthal) [reply content/productivity/custom-distribute-silver/plugin.js:1]: Also the timer " by default runs ever 30 seconds" etc

Comment (jacobrosenthal) [reply content/productivity/custom-distribute-silver/plugin.js:1]: Also please explain how the plugin works and the pair list. I believe the plugin works not automatically, but rather by you selecting a single asteroid and then selecting who it should send stuff to. This is called the pair list? Which can be 'backed up' by downloading pair list, and then 'uploading' it when you restart the browser and the plugin?

**Comment by thickforest** [comment]
> I think the screenshot size here is also a bit too big at 135 kb, to quote jaconrosenthal: `your screenshot is 200kb and thats really large, can you get that down to under 50k if possible?`

Thanks for your review, I have adjusted the screenshot size.

File: content/productivity/custom-distribute-silver/plugin.js (hunk 2 above)
Comment (thickforest) [reply content/productivity/custom-distribute-silver/plugin.js:1]: I have add comment at the top of the file.
Thanks for your review.

File: content/productivity/custom-distribute-silver/plugin.js [hunk 3]
```diff
@@ -0,0 +1,456 @@
...
+    // downloadButton
+    const downloadButton = document.createElement("button");
+    downloadButton.innerText = "Download Pairlist";
+    downloadButton.style.marginRight = "10px";
+    downloadButton.style.marginBottom = '10px';
+    downloadButton.addEventListener("click", () => {
+      let pairs = JSON.stringify(this.pairList);
+      var blob = new Blob([pairs], { type: 'application/json' }),
+      anchor = document.createElement('a');
+      anchor.download = df.getContractAddress().substring(0, 6) + '_pairs.json';
```
Comment (Stx69) [inline content/productivity/custom-distribute-silver/plugin.js:237]: anchor.download = new Date().toLocaleString() + "_" + df.getAccount() + '_Silver-pairs.json';

Comment (Stx69) [inline content/productivity/custom-distribute-silver/plugin.js:237]: anchor.download = new Date().toLocaleString() + "_" + df.getAccount() + '_Silver-pairs.json';

I like this format of output. More flexibile for more variants. Why is originaly df.getContractAddress?

**Review by Stx69 — COMMENTED** [review]
Hi, for me working and looks safe its a robust and comlex solution. Some thinks to learn again. 👍  Just the output format for me is not really the best but is a question.... 🍡

**Review by Stx69 — COMMENTED** [review]
Hello, small sugestion.

**Review by Stx69 — APPROVED** [review]

File: content/productivity/custom-distribute-silver/plugin.js (hunk 3 above)
Comment (thickforest) [reply content/productivity/custom-distribute-silver/plugin.js:237]: Thanks very much for your review and advise.
Output file with date can avoid duplicating the name when downloading.
`df.getAccount()` is more readability and flexibile than `df.getContractAddress()`.

File: content/productivity/custom-distribute-silver/plugin.js [hunk 4]
```diff
@@ -0,0 +1,456 @@
...
+    container.appendChild(uploadButton);
+    container.appendChild(globalButton);
+    container.appendChild(autoSecondsInfo);
+    container.appendChild(autoSecondsStepper);
+    container.appendChild(autoLabel);
+    container.appendChild(autoSilverCheck);
+  }
+
+  destroy() {
+    if (this.sendTimer) {
```
[the remark this answers is not in the document]
Comment (thickforest) [reply content/productivity/custom-distribute-silver/plugin.js]: Thanks for your review

**Pull request merged**